homeassistant/components/light/yeelight.py

```diff
@@ -0,0 +1,123 @@
+"""
+Support for Xiaomi Yeelight Wifi color bulb.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.yeelight/
+"""
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICES, CONF_NAME
+from homeassistant.components.light import (ATTR_BRIGHTNESS, ATTR_RGB_COLOR,
+                                            SUPPORT_BRIGHTNESS,
+                                            SUPPORT_RGB_COLOR, Light,
+                                            PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pyyeelight==1.0-beta']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'yeelight'
+
+SUPPORT_YEELIGHT = (SUPPORT_BRIGHTNESS | SUPPORT_RGB_COLOR)
+
+DEVICE_SCHEMA = vol.Schema({vol.Optional(CONF_NAME): cv.string, })
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA}, })
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Yeelight bulbs."""
+    lights = []
+    for ipaddr, device_config in config[CONF_DEVICES].items():
+        device = {'name': device_config[CONF_NAME], 'ipaddr': ipaddr}
+        lights.append(YeelightLight(device))
+
+    add_devices(lights)
+
+
+class YeelightLight(Light):
+    """Representation of a Yeelight light."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, device):
+        """Initialize the light."""
+        import pyyeelight
+
+        self._name = device['name']
+        self._ipaddr = device['ipaddr']
+        self.is_valid = True
+        self._bulb = None
+        try:
+            self._bulb = pyyeelight.YeelightBulb(self._ipaddr)
+        except socket.error:
+            self.is_valid = False
+            _LOGGER.error("Failed to connect to bulb %s, %s", self._ipaddr,
+                          self._name)
+
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return "{}.{}".format(self.__class__, self._ipaddr)
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._bulb.is_on()
```

Comment:
Properties should not make requests to devices, instead, these values should be cached on the instance within the `update` method.